### PR TITLE
Update and add more jsdoc for server, sslUtil and middleware

### PR DESCRIPTION
**Pull request by svbender**

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -19,15 +19,15 @@ const fsInterface = ui5Fs.fsInterface;
...
- * @param      {string}   options.port  Port to listen to
- * @param      {boolean}  [options.changePortIfInUse=false]  If true, change the port if it is already in use
- * @param      {string}   [options.protocol=http]  Protocol to be used (http|https|h2) - defaults to <pre>http</pre>
- * @param      {string}   [options.key]  Path to private key to be used for https
- * @param      {string}   [options.cert]  Path to certificate to be used for for https
-
- * @returns     {Promise}  Promise resolving with an object containing the <pre>port</pre> once the server is listening
+ * @param {Object} tree A (sub-)tree
+ * @param {Object} options Options
+ * @param {string} options.port Port to listen to
```
Comment (matz3) [inline lib/server.js]: Should be `number`, right?

File: lib/server.js [hunk 2]
```diff
@@ -19,15 +19,15 @@ const fsInterface = ui5Fs.fsInterface;
...
+ * @param {boolean} [options.changePortIfInUse=false] If true, change the port if it is already in use
+ * @param {string} [options.h2=false] Protocol to be used - defaults to <code>http</code>
```
Comment (matz3) [inline lib/server.js]: "Whether HTTP/2 should be used"

File: lib/server.js [hunk 3]
```diff
@@ -19,15 +19,15 @@ const fsInterface = ui5Fs.fsInterface;
...
+ * @param {string} [options.key] Path to private key to be used for https
+ * @param {string} [options.cert] Path to certificate to be used for for https
+ * @param {boolean} [acceptRemoteConnections=false] If true, listens to remote connections and not only to localhost connections
```
Comment (matz3) [inline lib/server.js]: `[options.acceptRemoteConnections=false]`

File: lib/server.js [hunk 4]
```diff
@@ -19,15 +19,15 @@ const fsInterface = ui5Fs.fsInterface;
...
- * @param      {string}   [options.protocol=http]  Protocol to be used (http|https|h2) - defaults to <pre>http</pre>
- * @param      {string}   [options.key]  Path to private key to be used for https
- * @param      {string}   [options.cert]  Path to certificate to be used for for https
-
- * @returns     {Promise}  Promise resolving with an object containing the <pre>port</pre> once the server is listening
+ * @param {Object} tree A (sub-)tree
+ * @param {Object} options Options
+ * @param {number} options.port Port to listen to
+ * @param {boolean} [options.changePortIfInUse=false] If true, change the port if it is already in use
+ * @param {string} [options.h2=false] Whether HTTP/2 should be used - defaults to <code>http</code>
```
Comment (RandomByte) [inline lib/server.js]: Not a string but a boolean

File: lib/server.js [hunk 5]
```diff
@@ -19,15 +19,15 @@ const fsInterface = ui5Fs.fsInterface;
...
+ * @param {string} [options.key] Path to private key to be used for https
+ * @param {string} [options.cert] Path to certificate to be used for for https
+ * @param {boolean} [options.acceptRemoteConnections=false] If true, listens to remote connections and not only to localhost connections
+ * @returns {Promise<Object>} Promise resolving with an object containing the <code>port</code> once the server is listening
```
Comment (RandomByte) [inline lib/server.js]: Promise resolving once the server is listening. It resolves with an object containing the <code>port</code>, <code>h2</code>-flag and a <code>close</code> function, which can be used to stop the server.

**Comment by RandomByte** [comment]
Very nice 🎉

**Pull request merged**